wit/types.wit

```diff
@@ -16,29 +25,29 @@ interface messaging-types {
     /// The interface doesn't highlight this difference in the type itself as that's uniquely a consumer issue.
     type channel = string;
   
-    /// Configuration includes a required list of channels the guest is subscribing to, and an optional list of extensions key-value pairs 
-    /// (e.g., partitions/offsets to read from in Kafka/EventHubs, QoS etc.).
+    /// Configuration includes a required list of channels the guest is subscribing to, and an
+    /// optional list of extensions key-value pairs (e.g., partitions/offsets to read from in 
+    /// Kafka/EventHubs, QoS etc.).
     record guest-configuration {
         channels: list<channel>,
         extensions: option<list<tuple<string, string>>>
     }
   
-    /// Format specification for messages 
-    ///  - more info: https://github.com/clemensv/spec/blob/registry-extensions/registry/spec.md#message-formats
-    ///  - message metadata can further decorate w/ things like format version, and so on.
-    enum format-spec {
-        cloudevents,
-        http,
-        amqp,
-        mqtt,
-        kafka,
-        raw
-    }
-  
-    /// A message with a binary payload, a format specification, and decorative metadata.
+    /// A message with a binary payload and additional information
     record message {
```

Comment:
I wonder if `message` should be a `resource` so that its contents are not always eagerly copied in full (for the same reason as `request` and `response` in WASI HTTP), for the benefit of cases where messages are simply forwarded or only partially inspected.  Also, the `complete-message`/`abandon-message` functions take a `message` by value which again does a full copy -- instead I think these functions want to be methods of a `message` resource (which has the benefit of giving messages a clear identity (for purposes of completion/abandonment) -- as it is, without any sort of "message id" field, I'm not sure how `complete-message`/`abandon-message` would be implemented.

Reply:
+1 for a `message` `resource`.

Reply:
I definitely think we should have the complete/abandon functionality be a resource. As for the body of the message, I think having that be behind a resource is good, but I would love to avoid streams if at all possible. Similarly to keyvalue, these values should be small as large values are generally seen as an antipattern. I think both Kafka and NATS have a 1MB default upper limit, but I assume many of the others are also likely to be small. 

With that in mind, would we want a way to "consume" the message when reading the body? I'm not sure what the proper thing here is to do with resources

Reply:
I agree you probably don't need a stream (at least in a 0.2 timeframe).  I'd suggest a resource roughly of the shape:
```wit
resource message {
  topic: func() -> channel;
  ...
  data: func() -> list<u8>;
}
```
(Because we're not trying to stream message bodies, I think we don't need the "may only call once" rules with wasi-http's `body.consume`.)